drivers/modem/modem_receiver.c

```diff
@@ -119,7 +116,9 @@ int mdm_receiver_recv(struct mdm_receiver_context *ctx,
 		return -EINVAL;
 	}
 
-	return k_pipe_get(&ctx->uart_pipe, buf, size, bytes_read, 1, K_NO_WAIT);
+	*bytes_read = ring_buf_get(&ctx->rx_rb, buf, size);
+
+	return 0;
```

Comment:
This is a documentation issue, but should be:
return (*bytes_read == 0 ? -EIO : 0);

Otherwise, we'll endless loop here waiting for < 0 result (meaning buffer is empty)
See: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/modem/wncm14a2a.c#L1081

Reply:
Of course I could change WNCM14A2A code to check for 0 bytes read and that might solve this issue as well.

Reply:
great idea